tests/pass/libc.rs

```diff
@@ -1,17 +1,142 @@
 // ignore-windows: No libc on Windows
 // compile-flags: -Zmiri-disable-isolation
 
+#![feature(io_error_more)]
 #![feature(rustc_private)]
 
 extern crate libc;
 
-#[cfg(any(target_os = "linux", target_os = "freebsd"))]
 fn tmp() -> std::path::PathBuf {
     std::env::var("MIRI_TEMP")
         .map(std::path::PathBuf::from)
         .unwrap_or_else(|_| std::env::temp_dir())
```

Comment:
This should probably be adjusted like the one in `fs.rs`. That might help with the Windows test failure.

It might not be enough tough -- this could be a tricky issue to resolve. But let's wee what happens when `tmp` at least takes care of the (back)slashes.